examples/parallel-coordinates/manipulable-axis.js

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+import {AbstractSeries} from 'react-vis';
+import {range} from 'd3-array';
+import {scaleLinear} from 'd3-scale';
+import {format} from 'd3-format';
+
+const predefinedClassName = 'rv-xy-manipulable-axis rv-xy-plot__axis';
+
+const animatedProps = [
+  'xRange', 'yRange', 'xDomain', 'yDomain',
+  'width', 'height', 'marginLeft', 'marginTop', 'marginRight', 'marginBottom',
+  'tickSize', 'tickTotal', 'tickSizeInner', 'tickSizeOuter'
+];
+
+function generatePoints({axisStart, axisEnd, numberOfTicks, axisDomain}) {
+  let left = Math.min(axisStart.x, axisEnd.x);
+  let right = Math.max(axisStart.x, axisEnd.x);
+  // generate the linear projection of the axis direction
+  let slope = (axisStart.y - axisEnd.y) / (axisStart.x - axisEnd.x);
+  let offset = axisStart.y - slope * axisStart.x;
+  // address the special case when the slope is infinite
+  if (axisStart.x === axisEnd.x) {
+    left = Math.min(axisStart.y, axisEnd.y);
+    right = Math.max(axisStart.y, axisEnd.y);
+    slope = 0;
+    offset = axisStart.x;
+  }
+  // construct a linear band of points, then map them
+  const pointSlope = (right - left) / (numberOfTicks);
+  const axisScale = scaleLinear().domain([left, right]).range(axisDomain.sort());
+
+  return {
+    slope: axisStart.x === axisEnd.x ? Infinity : slope,
+    points: range(left, right + pointSlope, pointSlope)
+      // this may be wrong for other directions, that remains to be seen
+      .map(val => ({y: val, x: slope * val + offset, text: axisScale(val)}))
+  };
+}
+
+function renderTicks(props) {
+  const {axisDomain, numberOfTicks, axisStart, axisEnd, tickValue} = props;
+  const {points} = generatePoints({axisStart, axisEnd, numberOfTicks, axisDomain});
+
+  return points
+    .map((point, index) => {
+      const tickProps = {
```

Comment:
in order to productionalize this component, this needs to manipuable from the props, specifically it needs to offer the length and automatically find the tangent to the slope